instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -94,17 +94,19 @@ def process_request(self, request):
 
         tracer = get_tracer(__name__, __version__)
 
-        attributes = collect_request_attributes(environ)
-
         span = tracer.start_span(
             self._get_span_name(request),
             kind=SpanKind.SERVER,
-            attributes=attributes,
             start_time=environ.get(
                 "opentelemetry-instrumentor-django.starttime_key"
             ),
         )
 
+        if span.is_recording():
```

Comment:
I wonder if it would be better to wrap the start_span in the check for is_recording here to avoid having to iterate through each attribute element, this might be an optimization though.

Reply:
How can this be done? `start_span` is the function that decides whether the returned span is going to be `is_recording` and returns the actual span.

Reply:
Hah, yeah of course... 🤦